include/ncengine/ecs/Registry.h

```diff
@@ -141,19 +159,19 @@ auto Registry::Add(EntityInfo info) -> Entity
     return handle;
 }
 
-template<std::derived_from<ComponentBase> T, class... Args>
+template<PooledComponent T, class... Args>
```

Comment:
All the functions here are the same logic from before, they just go from if constexprs to two separate functions.